Week2/todo_app/program.js

```diff
@@ -22,38 +23,91 @@ class TodoModel {
+        this.dbConnection.query(queryString, (err, results, fields) => {
+            if (err) {
+                callback(err);
+                return;
+            } else {
+                callback(null, results)
+            }
+        });
     }
 
     update(id, description, callback) {
-        // Write code and query to update and existing TODO item
+        const queryString = `UPDATE todo_items SET text = '${description}' WHERE id = ${id}`
+        this.dbConnection.query(queryString, (err, results, fields) => {
+            if (err) {
+                callback(err);
+                return;
+            } else {
+                callback(null, results)
+            }
+        })
     }
 
     delete(id, callback) {
-        // Write code and query to delete an existing TODO item
+        const queryString = ` DELETE FROM todo_items WHERE id = ${id}`;
+        this.dbConnection.query(queryString, (err, results, fields) => {
+            if (err) {
+                callback(err);
+                return;
+            } else {
+                callback(null, results)
+            }
+        })
     }
 
     tagTodoItem(todoItemId, tagId, callback) {
-        // Write code and query add a tag to a TODO item
+        const queryString = `INSERT INTO todo_item_tag(todo_item_id, tag_id) VALUES (${todoItemId}, ${tagId})`;
+        this.dbConnection.query(queryString, (err, results, fields) => {
+            if (err) {
+                callback(err);
+                return;
+            } else {
+                callback(null, results)
+            }
+        })
     }
-        
+
     untagTodoItem(todoItemId, tagId, callback) {
-        // Write code and query remove a tag from a TODO item
+        const queryString = `DELETE FROM todo_item_tag WHERE todo_item_id = ${todoItemId}`;
+        this.dbConnection.query(queryString, (err, results, fields) => {
+            if (err) {
+                callback(err);
+                return;
+            } else {
+                callback(null, results)
+            }
+        })
     }
 
     markCompleted(todoItemId, callback) {
-        // Write code to mark a TODO item as completed
+        const queryString = `UPDATE todo_items SET is_completed = 1 WHERE id = ${todoItemId}`;
+        this.dbConnection.query(queryString, (err, results, fields) => {
+            if (err) {
+                callback(err);
+                return;
+            } else {
+                callback(null, results)
+            }
+        })
     }
 }
 
 const dbConnection = mysql.createConnection({
-    host     : 'localhost',
-    user     : 'root',
-    password : '',
-    database : 'todo_app'
+    host: config.host,
+    user: config.user,
+    password: config.password,
+    database: 'todo_app'
 });
 
-dbConnection.connect(function(err) {
+const cmd = process.argv[2];
+const args = process.argv[3];
+const args2 = process.argv[4];
```

Comment:
All of your queries are vulnerable to SQL injection. You should try and, for each query, use escaping or prepared statements to prevent a SQL injection attack.